### PR TITLE
Deprecate Stripe Reader Revenue separate platform

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Unifies on-site Reader Revenue platforms, so that "Newspack" is the only one. 

This is a breaking change. Recurring donations will no longer result in Stripe Subscriptions – instead, WC Subscriptions will be created. If Stripe was set as the Reader Revenue platform, it will be changed to Newspack.  

If Stripe is set as the RR platform on a site, with these changes there will be consequences:

1. New recurring donations will result in WC Subscriptions instead of native Stripe Subscriptions
2. Existing Stripe Subscriptions will no longer be synchronised as orders to WC

### How to test the changes in this Pull Request:

1. On `master`, configure Stripe as the Reader Revenue platform
3. Switch to this branch
4. Observe the Reader Revenue platform in the Reader Revenue wizard is now set to "Newspack"
6. Make a recurring donation and then a one-time donation through the Donate block
7. Observe both processed in Stripe (in [Stripe Dashboard](https://dashboard.stripe.com/dashboard)) as single transactions 
8. Observe the recurring donation has resulted in a WC Subscription, and the one-time donation in a single order

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by dkoo** [comment]
@adekbadek time to update this one to the latest `master`?

**Review by dkoo — CHANGES_REQUESTED** [review]
Saw some issues and fatals during testing that I think could be resolved after fixing conflicts created by #2821. Also want to make sure that any code we're removing in this and https://github.com/Automattic/newspack-blocks/pull/1638 that are still used by our subscription migration tools are fully migrated to that repo. I see some methods and constants being removed here that are still referenced from over there.

File: includes/reader-revenue/woocommerce/class-woocommerce-connection.php [hunk 1]
```diff
@@ -114,29 +76,6 @@ public static function disable_woocommerce_setup() {
...
-	/**
-	 * Get a WC_Order_Item related to the product with the given SKU, or a Newspack donation product based on frequency.
-	 *
-	 * @param string      $frequency Frequency of the order's recurrence.
-	 * @param number      $amount Donation amount.
-	 * @param null|string $product_sku Product's SKU string, or null to get a Newspack donation product.
-	 *
-	 * @return boolean|WC_Order_Item_Product Order item product, or false if there's no product with matching SKU.
-	 */
-	public static function get_order_item( $frequency = 'once', $amount = 0, $product_sku = null ) {
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply includes/reader-revenue/woocommerce/class-woocommerce-connection.php:123]: Added to migration tools PR 👌

File: includes/reader-revenue/woocommerce/class-woocommerce-connection.php [hunk 2]
```diff
@@ -53,23 +35,13 @@ public static function init() {
 		\add_filter( 'woocommerce_email_enabled_customer_completed_order', [ __CLASS__, 'send_customizable_receipt_email' ], 10, 3 );
 		\add_action( 'cli_init', [ __CLASS__, 'register_cli_commands' ] );
 
+		\add_action( 'woocommerce_order_status_changed', [ __CLASS__, 'handle_order_status_change' ], 10, 4 );
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply includes/reader-revenue/woocommerce/class-woocommerce-connection.php]: Removed in bf349a64ce122a9c78edc3ec6eadb261c41441a3

File: includes/reader-revenue/woocommerce/class-woocommerce-connection.php [hunk 3]
```diff
@@ -1216,6 +408,38 @@ public static function force_allow_failed_payment_retry( $should_retry ) {
...
+		}
+		$frequency = 'once';
+		if ( function_exists( 'wcs_get_subscriptions_for_order' ) ) {
+			$related_subscriptions = \wcs_get_subscriptions_for_order( $order );
+			// In theory, there should be just one subscription per renewal.
+			$subscription = reset( $related_subscriptions );
+			$frequency    = $subscription->get_billing_period();
+		}
+
+		\Newspack\Google_Services_Connection::send_custom_event(
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply includes/reader-revenue/woocommerce/class-woocommerce-connection.php]: Removed in bf349a64ce122a9c78edc3ec6eadb261c41441a3

File: includes/reader-revenue/woocommerce/class-woocommerce-connection.php [hunk 4]
```diff
@@ -1216,6 +408,38 @@ public static function force_allow_failed_payment_retry( $should_retry ) {
...
+		// Only run when order is completed.
+		if ( 'completed' !== $to ) {
+			return;
+		}
+		$frequency = 'once';
+		if ( function_exists( 'wcs_get_subscriptions_for_order' ) ) {
+			$related_subscriptions = \wcs_get_subscriptions_for_order( $order );
+			// In theory, there should be just one subscription per renewal.
+			$subscription = reset( $related_subscriptions );
+			$frequency    = $subscription->get_billing_period();
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply includes/reader-revenue/woocommerce/class-woocommerce-connection.php]: Yes, removed in bf349a64ce122a9c78edc3ec6eadb261c41441a3

File: includes/configuration_managers/class-woocommerce-configuration-manager.php [hunk 5]
```diff
@@ -101,32 +101,6 @@ function( $gateway ) {
...
-		$stripe_data = [
-			'enabled'                     => 'yes' === $stripe->get_option( 'enabled', false ) ? true : false,
-			'testMode'                    => 'yes' === $stripe->get_option( 'testmode', false ) ? true : false,
-			'publishableKey'              => $stripe->get_option( 'publishable_key', '' ),
-			'secretKey'                   => $stripe->get_option( 'secret_key', '' ),
-			'testPublishableKey'          => $stripe->get_option( 'test_publishable_key', '' ),
-			'testSecretKey'               => $stripe->get_option( 'test_secret_key', '' ),
-			'allow_covering_fees'         => get_option( 'newspack_donations_allow_covering_fees', true ),
-			'allow_covering_fees_default' => get_option( 'newspack_donations_allow_covering_fees_default', false ),
-			'allow_covering_fees_label'   => get_option( 'newspack_donations_allow_covering_fees_label', '' ),
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply includes/configuration_managers/class-woocommerce-configuration-manager.php:125]: Good catch! Added in 83f627675e6a38ab1c55a0edbb1d8402fa971b24

**Review by dkoo — APPROVED** [review]
Looking good now and working well!

**Pull request merged**